### ISSUE TITLE
Unable to detect git repo in 1.37

### ISSUE THREAD

**Issue by naveensrinivasan** [golangci/golangci-lint#1941]
- [x] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [x] Yes, I've searched similar issues on GitHub and didn't find any.
- [x] Yes, I've included all information below (version, config, etc).
- [ ] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

<details><summary>Description of the problem</summary>

Upgrading from 1.36 to 1.37 is causing `new` issues checks fails.

`level=warning msg="[runner] Can't process result by diff processor: can't prepare diff by revgrep: could not read git repo: error executing git diff HEAD~: exit status 128" `

</details>

<details><summary>Version of golangci-lint</summary>

```console
$ golangci-lint --version
# paste output here
1.37
```

</details>

<details><summary>Config file</summary>

```console
$ cat .golangci.yml
# paste output here
```

</details>

<details><summary>Go environment</summary>

```console
$ go version && go env
# paste output here
 1.16
```

</details>

<details><summary>Verbose output of running</summary>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
# paste output here
```

</details>

<details><summary>Code example or link to a public repository</summary>

https://github.com/ossf/scorecard/pull/369

https://github.com/ossf/scorecard/runs/2461368761?check_suite_focus=true

</details>

**Comment by ldez** [comment]
Hello,

From your logs:

```
Run actions/checkout@v2
  with:
    repository: ossf/scorecard
    token: ***
    ssh-strict: true
    persist-credentials: true
    clean: true
    fetch-depth: 1
    lfs: false
    submodules: false
```
https://github.com/ossf/scorecard/runs/2461368761?check_suite_focus=true

I think that `fetch-depth: 1` not enough to create the right diff.

I think you have to do what I have explain here https://github.com/golangci/golangci-lint/issues/1088#issuecomment-801540792

```yml
    steps:

      - name: Check out code
        uses: actions/checkout@v2
        with:
          fetch-depth: 0
```

**Comment by ldez** [comment]
In v1.37.0 (#1723) we fixed #970: the problem was a silent failure on large git histories.
Maybe before the update, golangci-lint skipped the analysis on your repo.

**Comment by naveensrinivasan** [comment]
> In v1.37.0 (#1723) we fixed #970: the problem was a silent failure on large git histories.
> Maybe before the update, golangci-lint skipped the analysis on your repo.

Thank you! That solves the issue 👍
